src/decorators.ts

```diff
@@ -44,7 +49,18 @@ export function property(options?: PropertyOptions) {
     const reflectToAttribute: boolean =
         options && options.reflectToAttribute || false;
     const readOnly: boolean = options && options.readOnly || false;
-    const type = Reflect.getMetadata('design:type', proto, propName);
+
+    let type = options && options.type;
+    if (!type) {
+      if (Reflect && Reflect.getMetadata) {
+        type = Reflect.getMetadata('design:type', proto, propName);
+      } else {
+        console.error(
+            'Property ${propName} has no type ' +
```

Comment:
I don't think it's an error to not have a type on a property, at least not a runtime error.

Reply:
Got rid of the warning. Also now check `window.Reflect` instead of just `Reflect` since IE and others won't have any Reflect global at all.

Reply:
That is a good point actually... You may have changed my mind

Reply:
😛

Reply:
Ok, restored the warning, but made it so you'll only get the warning if both 1. You didn't explicitly set the type field (where undefined counts as set), and 2. The reflect polyfill isn't loaded, or metadata wasn't injected (where undefined counts as injected).